scripts/js/charts.js

```diff
@@ -152,169 +122,217 @@ const customTooltips = function (context) {
   tooltipEl.classList.remove("left", "right", "center", "top", "bottom");
   tooltipEl.classList.add(tooltip.xAlign, tooltip.yAlign);
+}
 
-  // Set Text
-  if (tooltip.body) {
-    const titleLines = tooltip.title || [];
-    const bodyLines = tooltip.body.map(bodyItem => bodyItem.lines);
-    let innerHtml = "<thead>";
+function setTooltipContent(tooltipEl, tooltip) {
+  const bodyLines = tooltip.body.map(bodyItem => bodyItem.lines);
+  if (bodyLines.length === 0) return;
 
-    for (const title of titleLines) {
-      innerHtml += "<tr><th>" + title + "</th></tr>";
-    }
+  const titleLines = tooltip.title || [];
+  let tooltipHtml = "<thead>";
 
-    innerHtml += "</thead><tbody>";
-    let printed = 0;
-
-    const devicePixel = (1 / window.devicePixelRatio).toFixed(1);
-    for (const [i, body] of bodyLines.entries()) {
-      const labelColors = tooltip.labelColors[i];
-      let style = "background-color: " + labelColors.backgroundColor;
-      style += "; outline: 1px solid " + labelColors.backgroundColor;
-      style += "; border: " + devicePixel + "px solid #fff";
-      const span = "<span class='chartjs-tooltip-key' style='" + style + "'></span>";
-
-      const num = body[0].split(": ");
-      // do not display entries with value of 0 (in bar chart),
-      // but pass through entries with "0.0% (in pie charts)
-      if (num[1] !== "0") {
-        innerHtml += "<tr><td>" + span + body + "</td></tr>";
-        printed++;
-      }
-    }
+  for (const title of titleLines) {
+    tooltipHtml += `<tr><th>${title}</th></tr>`;
+  }
 
-    if (printed < 1) {
-      innerHtml += "<tr><td>No activity recorded</td></tr>";
+  tooltipHtml += "</thead><tbody>";
+
+  const devicePixel = (1 / window.devicePixelRatio).toFixed(1);
+  let printed = 0;
+
+  for (const [i, body] of bodyLines.entries()) {
+    const labelColors = tooltip.labelColors[i];
+    const style =
+      `background-color: ${labelColors.backgroundColor}; ` +
+      `outline: 1px solid ${labelColors.backgroundColor}; ` +
+      `border: ${devicePixel}px solid #fff`;
+    const span = `<span class="chartjs-tooltip-key" style="${style}"></span>`;
+
+    const num = body[0].split(": ");
+    // Do not display entries with value of 0 in bar chart,
+    // but pass through entries with "0.0%" (in pie charts)
+    if (num[1] !== "0") {
+      tooltipHtml += `<tr><td>${span}${body}</td></tr>`;
+      printed++;
     }
+  }
 
-    innerHtml += "</tbody>";
-
-    const tableRoot = tooltipEl.querySelector("table");
-    tableRoot.innerHTML = innerHtml;
+  if (printed < 1) {
+    tooltipHtml += "<tr><td>No activity recorded</td></tr>";
   }
 
-  const canvasPos = this.chart.canvas.getBoundingClientRect();
+  tooltipHtml += "</tbody>";
+
+  const tableRoot = tooltipEl.querySelector("table");
+  tableRoot.innerHTML = tooltipHtml;
+}
+
+function positionTooltip(tooltipEl, tooltip, context) {
+  if (tooltip.opacity === 0 || tooltipEl.style.opacity === 0) return;
```

Comment:
Yet sometimes it seems we are allowed to just do classic simple if statements

If we're doing it above, why doesn't this line have something like `const isTooltipOpaque = (tooltiip.opacity.......` above it?